src/simtools/production_configuration/generate_production_grid.py

```diff
@@ -0,0 +1,384 @@
+        -------
+        numpy.ndarray
+            Array of axis values following a power-law distribution.
+        """
+        lin_space = np.linspace(0, 1, binning)
+        lin_space = np.clip(lin_space, 1e-10, 1 - 1e-10)  # Avoid division by zero
+        return axis_range[0] + (axis_range[1] - axis_range[0]) * (lin_space) ** (power_law_index)
+
+    def convert_altaz_to_radec(self, alt, az):
+        """
+        Convert Altitude/Azimuth (AltAz) coordinates to Right Ascension/Declination (RA/Dec).
+
+        Parameters
+        ----------
+        alt : float
+            Altitude angle in degrees.
+        az : float
+            Azimuth angle in degrees.
+
+        Returns
+        -------
+        SkyCoord
+            SkyCoord object containing the RA/Dec coordinates.
+        """
+        alt_rad = alt.to(u.rad)
+        az_rad = az.to(u.rad)
+        aa = AltAz(
+            alt=alt_rad,
+            az=az_rad,
+            location=self.observing_location,
+            obstime=self.observing_time,
+        )
+        skycoord = SkyCoord(aa)
+        return skycoord.icrs  # Return RA/Dec in ICRS frame
+
+    def convert_coordinates(self, grid_points: list[dict]) -> list[dict]:
+        """
+        Convert the grid points to RA/Dec coordinates if necessary.
+
+        Parameters
+        ----------
+        grid_points : list of dict
+            List of grid points, each represented as a dictionary with axis
+              names as keys and values.
+
+        Returns
+        -------
+        list of dict
+            The grid points with converted RA/Dec coordinates.
+        """
+        if self.coordinate_system == "ra_dec":
+            for point in grid_points:
+                if "zenith_angle" in point and "azimuth" in point:
+                    alt = (90.0 * u.deg) - point.pop("zenith_angle")
+                    az = point.pop("azimuth")
+                    radec = self.convert_altaz_to_radec(alt, az)
+                    point["ra"] = radec.ra.deg * u.deg
+                    point["dec"] = radec.dec.deg * u.deg
+        return grid_points
+
+    def clean_grid_output(self, grid_points, output_file=None):
+        """Clean the grid output and save to a file or print to the console."""
+        cleaned_points = []
+
+        for point in grid_points:
+            cleaned_point = {}
+            for key, value in point.items():
+                if isinstance(value, dict):
+                    # Nested dictionaries
+                    cleaned_point[key] = {k: self.serialize_quantity(v) for k, v in value.items()}
+                else:
+                    cleaned_point[key] = self.serialize_quantity(value)
+
+            cleaned_points.append(cleaned_point)
+
+        output_data = json.dumps(cleaned_points, indent=4)
+
+        if output_file:
+            with open(output_file, "w", encoding="utf-8") as f:
+                f.write(output_data)
+            self._logger.info(f"Output saved to {output_file}")
```

Comment:
So close to covering all lines....why not these?

Reply:
Done